pkgs/compiler-test/tests/compiler/test/runtime.rkt

```diff
@@ -1,33 +1,53 @@
 #lang racket/base
 (require racket/system
+         racket/file
          compiler/find-exe)
 
 (define exe (find-exe))
 
-(define (try mode mod expect . extra-args)
+;; Return last line of the input, or #f if it's empty.
+(define (input-last-line input)
+  (for/fold ([prev #f]) ([s (in-lines input)])
+      (if (or (eof-object? s)
+              (equal? s "1 test passed"))
+          prev
+          s)))
+
+(define (try mode mod expect post-check . extra-args)
```

Comment:
One minor nit: there's really no need to create a new parameter `post-check`. You can reuse the parameter `expect`, since `expect` and `post-check` are completely disjoint. Then, you can check if `expect` is a procedure, and if it is, apply it to the output. Otherwise, use `equal?` to test it against the output.

Reply:
(And add a comment above that `expect` is either a procedure or a string.)

Reply:
Thanks, sorawee, fixed in the new commit.